packages/terra-functional-testing/src/reporters/spec-reporter/wdio-spec-reporter.js

```diff
@@ -0,0 +1,147 @@
+const fs = require('fs');
+const path = require('path');
+const WDIOReporter = require('@wdio/reporter').default;
+const getOutputDir = require('./get-output-dir');
+
+class SpecReporter extends WDIOReporter {
+  /**
+   * Hook invoked when the runner ends.
+   * @param {RunnerStats} runner - The test runner stats object.
+   */
+  onRunnerEnd(runner) {
+    // The root suite is always at index 0 and is guaranteed to exist.
+    const rootSuite = this.currentSuites[0];
+    const results = SpecReporter.formatResults(rootSuite, runner);
+
+    SpecReporter.writeResults(runner, results);
+  }
+
+  /**
+   * Formats the test runner spec results.
+   * @param {SuiteStats} runner - The root suite object.
+   * @param {RunnerStats} runner - The test runner stats object.
+   * @returns {Object} - A formatted results object for a single spec file.
+   */
+  static formatResults(rootSuite, runner) {
+    const {
+      start,
+      end,
+      capabilities,
+      specs,
+    } = runner;
+
+    const { suites, tests } = rootSuite;
+
+    return {
+      spec: specs[0],
+      start,
+      end,
+      capabilities: SpecReporter.formatCapabilities(capabilities),
+      packageName: SpecReporter.getPackageName(specs[0]),
+      suites: SpecReporter.formatSuites(suites),
+      tests: SpecReporter.formatTests(tests),
+    };
+  }
+
+  /**
+   * Formats the browser capabilities.
+   * @param {Object} capabilities - The browser capabilities.
+   */
+  static formatCapabilities(capabilities) {
+    const { browserName, platform, version } = capabilities;
+
+    return { browserName, platform, version };
+  }
+
+  /**
+   * Formats the test suite results.
+   * @param {array} suites - The collection of suites.
+   * @returns {Object} - A formatted suite results object.
+   */
+  static formatSuites(suites) {
+    return suites.map((suite) => {
+      const {
+        title,
+        fullTitle,
+        duration,
+        start,
+        end,
+        tests,
+      } = suite;
+
+      return {
+        title,
+        fullTitle,
+        duration,
+        start,
+        end,
+        tests: SpecReporter.formatTests(tests),
+        suites: SpecReporter.formatSuites(suite.suites),
+      };
+    });
+  }
+
+  /**
+   * Formats the test results.
+   * @param {array} tests - An array of test results.
+   * @returns {Object} - A formatted test results object.
+   */
+  static formatTests(tests) {
+    return tests.map((test) => {
+      const {
+        title,
+        fullTitle,
+        duration,
+        start,
+        end,
+        state,
+        error,
```

Comment:
The [error message/stack](https://gist.github.com/StephenEsser/6aa28588b17c9e313307133324221647#file-results-en-terra-default-theme-chrome-terra-functional-testing-json-L90-L91) in the sample output seems hard to read. Is it just a bad example or there is some formatting issue?

Reply:
It's formatted chalk'ed output so most of what you're seeing are special unicode characters. For now I think this is serialized as expected assuming it will be displayed in a terminal output. It would need to be sanitized if displayed outside of the terminal